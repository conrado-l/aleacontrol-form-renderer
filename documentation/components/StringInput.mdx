---
name: StringInput
---

import { Playground, PropsTable } from 'docz'
import StringInput from '../../src/components/StringInput.js'

# StringInput

<PropsTable of={StringInput} />

## Description
A simple text input
## Demo and usage

<Playground>
    <StringInput value='Hello' name='greeting' col={4} update={(e) => e}/>
</Playground>
