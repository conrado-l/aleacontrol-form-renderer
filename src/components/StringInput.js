import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Form from 'react-bootstrap/Form'

/**
 * Regular text type input
 */
class StringInput extends Component {
  constructor (props) {
    super(props)
    this.state = {
      value: props.value
    }

    this.handleChange = this.handleChange.bind(this)
  }

  /**
   * Updates the input value and sends the value through the parent callback function
   * @param {object} e Event
   */
  handleChange (e) {
    const newValue = e.target.value
    this.setState({ value: newValue })
    this.props.update(this.props.name, newValue)
  }

  render () {
    return (
      <Row>
        <Col xs={12} sm={this.props.col}>
          <Form.Control
            type='text'
            name={this.props.name}
            value={this.state.value || ''}
            onChange={this.handleChange} />
        </Col>
      </Row>
    )
  }
}

StringInput.propTypes = {
  value: PropTypes.string,
  name: PropTypes.string,
  col: PropTypes.number,
  update: PropTypes.func
}

export default StringInput
