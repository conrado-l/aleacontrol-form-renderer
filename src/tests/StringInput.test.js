import React from 'react'
import { mount } from 'enzyme'
import StringInput from '../components/StringInput'
import renderer from 'react-test-renderer'

describe('StringInput', () => {
  const mockupProps = {
    name: 'first_name',
    value: 'Paul'
  }
  const inputFn = jest.fn()

  it('should render correctly string component', () => {
    const StringInputComponent = renderer
      .create(<StringInput name={mockupProps.name} value={mockupProps.value} />)
      .toJSON()
    expect(StringInputComponent).toMatchSnapshot()
  })

  it('should render the input with the proper value and name', () => {
    const component = mount(<StringInput name={mockupProps.name} value={mockupProps.value} />)

    expect(
      component.containsMatchingElement(
        <input type='text' value={mockupProps.value} name={mockupProps.name} />
      )
    ).toBeTruthy()
  })

  it('should call the update callback function with proper parameters on input', () => {
    const StringInputComponent = mount(<StringInput name={mockupProps.name} value={mockupProps.value} update={inputFn} />)
    const inputString = 'Aleacontrol'

    StringInputComponent.find('input').simulate('change', {
      target: { value: inputString }
    })

    expect(inputFn).toHaveBeenCalledWith(mockupProps.name, inputString)
  })
})
