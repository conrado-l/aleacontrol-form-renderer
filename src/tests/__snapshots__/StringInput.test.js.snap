// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringInput should render correctly string component 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <input
      className="form-control"
      name="first_name"
      onChange={[Function]}
      type="text"
      value="Paul"
    />
  </div>
</div>
`;
